File: golang/Leetcode/14.LongestCommonPrefix.go
package Leetcode

func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}
	if len(strs) == 1 {
		return strs[0]
	}
	firstWord := strs[0]
	for i := 0; i < len(firstWord); i++ {
		firstPrefix := firstWord[i]
		for j := 1; j < len(strs); j++ {
			if i >= len(strs[j]) || strs[j][i] != firstPrefix {
				return firstWord[:i]
			}
		}
	}
	return firstWord
}

File: typescript/14. Longest Common Prefix.ts
function longestCommonPrefix(strs: string[]): string {
    if (!strs.length) return "";

    strs.sort();

    let firstArr: string[] = strs[0].split('');
    let lastArr: string[] = strs[strs.length - 1].split('');

    let result = "";

    for (let i = 0; i < firstArr.length; i++) {
        if (firstArr[i] === lastArr[i]) {
            result += firstArr[i];
        } else {
            break;
        }
    }
    return result;
}

// test